### PR TITLE
Add coverage check

### PR DESCRIPTION
**Pull request by job13er**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #none# - documentation fixes and/or test additions
 - [ ] #patch# - backwards-compatible bug fix
 - [x] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

# CHANGELOG
* **Added** the ability to ensure code coverage does not decrease because of a PR. Simply add your project's current code coverage in `package.json` under: 
  ```js
  "pr-bumper": {
    "coverage": 95.93
  }
   ``` 
  Then add a `pr-bumper check-coverage` call to your CI. The `check-coverage` command will fail if the current code coverage is below the baseline in your `package.json`.  
  
  **NOTE**: You must place the `pr-bumper check-coverage` line **after** you run your tests, or `pr-bumper` will not be able to find the current code coverage information to check against the baseline, and your build will fail regardless of your code coverage. 
  
  During a `pr-bumper bump` command, the current coverage will be saved to `package.json`

* **Added** the ability to customize the name of the changelog file to prepend (defaults to `CHANGELOG.md` still). 
* **Added** a `[pr-bumper]` prefix to all commits made by `pr-bumper` so they can be easily identified. 
* **Added** a `pr-bumper: ` prefix to all messages logged to the console by `pr-bumper` so they can be easily identified. 
* **Added** initial coverage baseline to `package.json` within this project to enable code coverage check. 
* **Added** `check-coverage` check after the build. 
* **Added** `.travis/maybe-check-coverage.sh` script to wrap a call to `pr-bumper check-coverage` and skip it if the commit being built is a `pr-bumper` commit.
* **Updated** how `.travis/is-bump-commit.sh` verifies that a commit came from `pr-bumper` to match the new prefix being used by `pr-bumper` for commits.

### CODE REVIEW

**Comment by job13er** [comment]
Looks like I broke node@5 support, looking at it now.

**Comment by job13er** [comment]
Here's what the CI build said about coverage (we should probably look at posting a comment instead of just logging it to console)

```
The command ".travis/maybe-test.sh" exited with 0.
$ .travis/maybe-check-coverage.sh
pr-bumper: No .pr-bumper.json found, using defaults
pr-bumper: pr-bumper::config: prNumber [86], isPr [true]
pr-bumper: Detected VCS provider: github 
pr-bumper: Detected CI provider: travis 
pr-bumper: Coverage increased 0.81% (from 91.24% to 92.05%) in this PR! Good job :)
```

**Comment by job13er** [comment]
NOTE: Please don't merge this until I'm online so I can watch the merge build and ensure it does what it's supposed to do, and fix it quickly if it does not :)

**Comment by job13er** [comment]
@sandersky self-merging to test the merge bit while no one else is online

**Comment by job13er** [comment]
Approved

File: .travis/is-bump-commit.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 isBumpCommit() {
   msg=`git log --pretty=format:'%s' -1`
-  if [ "$msg" = "Automated version bump" ]
+  if [[ "$msg" =~ ^\[pr-bumper\]* ]]
```
Comment (sandersky) [inline .travis/is-bump-commit.sh]: Could this cause problems if I do something like `git commit -m "fix pr-bumper config"` on a repo using this project?

Comment (job13er) [inline .travis/is-bump-commit.sh]: Nope, not unless you did
```
git commit -m "[pr-bumper] fix config"
```
It's checking that the msg starts with `[pr-bumper]` I included the brackets for just that reason.

Comment (sandersky) [inline .travis/is-bump-commit.sh]: Ah, somehow I overlook the square brackets. Yeah I see no reason to ever do that manually in a commit message.

**Pull request merged**